pkg/distributor/http.go

```diff
@@ -1,19 +1,34 @@
 package distributor
 
 import (
+	"encoding/json"
 	"net/http"
 
 	"github.com/cortexproject/cortex/pkg/util"
 
 	"github.com/grafana/loki/pkg/logproto"
 )
 
+var contentType = http.CanonicalHeaderKey("Content-Type")
+
+const applicationJSON = "application/json"
+
 // PushHandler reads a snappy-compressed proto from the HTTP body.
 func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 	var req logproto.PushRequest
-	if _, err := util.ParseProtoReader(r.Context(), r.Body, &req, util.RawSnappy); err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+
+	switch r.Header.Get(contentType) {
+	case applicationJSON:
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
```

Comment:
I dont see anywhere where we make sure that labels are correctly formated.  Do we need that?

Reply:
Thats enforced by the label parsing in the ingester.  Its a bit... weird.

Reply:
Yeah, I've seen a few folks asking about having it as a k:v structure rather than a string.  I started looking at how much would be involved to support both and found out it would be quite...involved.

Reply:
Lets file an issue for this and we can discuss further there.